runtime/lua/vim/lsp/_watchfiles.lua

```diff
@@ -264,15 +264,27 @@ function M.register(reg, ctx)
     end
   end
 
-  local watching = {}
-  for _, w in ipairs(watch_regs) do
-    if not watching[w.base_dir] then
-      watching[w.base_dir] = true
-      table.insert(
-        cancels[client_id][reg.id],
-        M._watchfunc(w.base_dir, { uvflags = { recursive = true } }, callback(w.base_dir))
-      )
+  for base_dir, watches in pairs(watch_regs) do
+    local include_patterns = {}
+    for _, w in ipairs(watches) do
+      for _, p in ipairs(w.pattern) do
+        table.insert(include_patterns, p)
```

Comment:
```suggestion
        include_patterns[#include_patterns + 1] =  p
```

Reply:
Out of curiosity, what's the difference between these two? There are a number of places that call `table.insert` like this so I'm wondering if those should be updated too.

Reply:
inline expression fast in `table.insert` in luajit when you do it in a for loop